homeassistant/components/snmp/switch.py

```diff
@@ -162,6 +162,15 @@ async def async_setup_platform(
     request_args = await async_create_request_cmd_args(
         hass, auth_data, UdpTransportTarget((host, port)), baseoid
     )
+    # Switches are read-only if command OID is not set which means they can't be turned on or off
+    # however since CONF_COMMAND_OID is marked as optional, we need to check if it is set
```

Comment:
My device is quite old and I'm not sure if it follows the SNMP specification completely, but in my case, the switches work correctly and as expected when they use the `baseoid` without a `command_oid` setting.

I think the previous logic to just use the `baseoid` if there's not `command_oid` configured will still work as expected. It would also prevent having to introduce a breaking change.

Reply:
That makes more sense.

Adjusted in 847d6df4b402be6072740e263a8bb0563e9da8d1